test/fluree/db/json_ld/api_test.cljc

```diff
@@ -113,29 +112,27 @@
          (let [conn         @(fluree/connect
                                {:method :file :storage-path storage-path
                                 :defaults
-                                {:context test-utils/default-context}})
+                                {:context test-utils/default-context
+                                 :context-type :keyword}})
```

Comment:
I want to make sure I'm understanding the role of `context-type` correctly. Is it the case that on conns, ledgers, and dbs:
- the default context is kept in string form internally
- the `context-type` controls which format the context will be in when you retrieve it (via protocol method)
- so the `context-type` stored at the default level is the default format you'll get upon said retrieval
- if you supply a `context-type` in the call to the protocol method, _that_ will determine what you get back?

So there are two defaults per "component" now:
- the context itself (its contents)
- the format in which you will receive context upon retrieval


All this would mean that weird combinations wouldn't matter, right? Like, say I create a conn with a default context and a `context-type` of `:keyword`, and then create a ledger with a `defaultContext` that's a string and I don't pass in a `context-type`. Is it the case that nothing will break because those `context-types` are about just-in-time formatting, not anything deeper?

Reply:
Correct, context-type is just-in-time formatting. All contexts are managed as strings internally.

If you want a keyword context (in CLJ), that conversion is always done at the point of the query or transaction against a DB (we cache it to make it efficient). So specifying it on a connection only acts as a default, so a lazy (me) CLJ dev doesn't have to specify this with every query.